theano/gpuarray/type.py

```diff
@@ -572,6 +592,8 @@ def gpuarray_shared_constructor(value, name=None, strict=False,
     See :func:`theano.shared`.
 
     """
+    if target is None and not move_to_gpu(value):
+        raise TypeError('We do not move that data by deault to the GPU')
```

Comment:
deault -> default

Also, maybe make target another unique object (like NoContext), since None is a valid context name.